src/operator/mshadow_op.h

```diff
@@ -1087,10 +593,10 @@ struct smooth_l1_gradient {
   template<typename DType>
   MSHADOW_XINLINE static DType Map(DType a, DType b) {
     b *= b;
-    if (a > DType(1.0f) / b) {
-      return DType(1.0f);
-    } else if (a < DType(-1.0f) / b) {
-      return DType(-1.0f);
+    if (a > 1.0f / b) {
```

Comment:
This is in the kernel; calculating `1.0f/b` twice is expensive. Maybe caching the result of `1.0f/b` can save some time.